eng/pipelines/templates/steps/analyze.yml

```diff
@@ -4,18 +4,60 @@ parameters:
     default: stable
 
 steps:
+  - ${{ if ne(parameters.Toolchain, 'nightly') }}:
+      - template: /eng/pipelines/templates/steps/use-rust.yml@self
+        parameters:
+          Toolchain: nightly
+
   - template: /eng/pipelines/templates/steps/use-rust.yml@self
     parameters:
       Toolchain: ${{ parameters.Toolchain }}
 
+  - template: /eng/common/pipelines/templates/steps/save-package-properties.yml
+
   - pwsh: |
+      . ./eng/common/scripts/common.ps1
       . ./eng/common/scripts/Helpers/CommandInvocation-Helpers.ps1
 
       $env:RUSTDOCFLAGS = "-D warnings"
       $env:RUSTFLAGS = "-Dwarnings"
 
       Invoke-LoggedCommand "cargo +$(Toolchain) check -p azure_core --no-default-features"
+
+      Write-Host "`n`n"
       Invoke-LoggedCommand "cargo +$(Toolchain) fmt --all -- --check"
+
+      Write-Host "`n`n"
       Invoke-LoggedCommand "cargo +$(Toolchain) clippy --all"
+
+      Write-Host "`n`n"
       Invoke-LoggedCommand "cargo +$(Toolchain) doc --all --no-deps"
+
+      # Verify package dependencies
+
+      $verifyDependenciesScript = Join-Path $RepoRoot '/eng/scripts/verify-dependencies.rs' -Resolve
+      $packageInfoPath = '$(Build.ArtifactStagingDirectory)/PackageInfo'
+
+      if ('$(NoPackagesChanged)' -ne 'true') {
```

Comment:
What is going on here? In PowerShell, this won't resolve. Single quotes are a literal, and even if it were double quotes `$(NoPackagesChanged)` wouldn't resolve a variable (`${NoPackagesChanged}`) would. `NoPackagesChanged` would then have to be a command.

Even after all that, why is it returning a boolean as a string? This makes truthy comparisons difficult and non-idiomatic to PowerShell.

All said, I don't think this will work the way you intend. This looks more like transpiled AzDO Pipelines grammar.

Reply:
It's because we're using PowerShell in YAML.   This is checking the string content of the pipeline variable `NoPackagesChanged`

Reply:
It'll compile to `if ('' -ne 'true')` or `if ('true' -ne 'true')` before invocation.

Reply:
This script is getting almost complex enough to put into a .ps1 file instead of inline.

Reply:
That's fine, but let's make sure it's clear it's not for devs' inner loop. We don't need scripts for people to learn when typical `cargo` commands should work just fine.